src/ember-app.js

```diff
@@ -492,20 +492,30 @@ class EmberApp {
       }
     }
 
-    debug('reading array of app file paths from manifest');
-    let appFiles = manifest.appFiles.map(function(appFile) {
-      return path.join(distPath, appFile);
-    });
+    let appFiles, vendorFiles, htmlFile, html;
 
-    debug('reading array of vendor file paths from manifest');
-    let vendorFiles = manifest.vendorFiles.map(function(vendorFile) {
-      return path.join(distPath, vendorFile);
-    });
+    if (manifest.htmlEntrypoint) {
+      let htmlEntrypoint = require('./html-entrypoint');
+      ({ appFiles, vendorFiles, html } = htmlEntrypoint(distPath, manifest.htmlEntrypoint));
+    } else {
```

Comment:
High level question: With this new API would addons be able to use `updateFastBootManifest` hook + embroider? 

If so, can we add a test for it?

Reply:
I don't understand how that test would be relevant to this repo. `fastboot` only consumes the manifest, it doesn't have anything to do with constructing the manifest.

(Embroider itself has test coverage to ensure that existing addons that use updateFastbootManifest work correctly, starting from https://github.com/embroider-build/embroider/blob/b88f660cc6a6629de916123318c909d5efd3d8a9/test-packages/fastboot-addon/index.js#L6)

Reply:
I understand that it only reads the config. I want to make sure the construct isn't broken. If it is in the embroider repo, I am fine with it.